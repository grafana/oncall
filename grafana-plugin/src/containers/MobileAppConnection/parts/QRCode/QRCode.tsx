import React, { FC } from 'react';

import { QRCodeSVG } from 'qrcode.react';

import { Block } from 'components/GBlock/Block';

type Props = {
  value: string;
  className?: string;
};

export const QRCode: FC<Props> = (props: Props) => {
  const { value, className = '' } = props;

  return (
    <Block bordered className={className}>
      <QRCodeSVG value={value} size={256} />
    </Block>
  );
};
