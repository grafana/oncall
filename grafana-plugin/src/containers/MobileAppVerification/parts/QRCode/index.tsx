import React, { FC } from 'react';

import QRCodeBase from 'react-qr-code';

import Block from 'components/GBlock/Block';

type Props = {
  value: string;
};

const QRCode: FC<Props> = ({ value }) => (
  <Block bordered>
    <QRCodeBase value={value} />
  </Block>
);

export default QRCode;
