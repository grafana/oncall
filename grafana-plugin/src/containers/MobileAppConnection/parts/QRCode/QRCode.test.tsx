import React from 'react';

import { render } from '@testing-library/react';

import { QRCode } from './QRCode';

describe('QRCode', () => {
  test('it renders properly', () => {
    const component = render(<QRCode value="helloooo" />);
    expect(component.container).toMatchSnapshot();
  });
});
