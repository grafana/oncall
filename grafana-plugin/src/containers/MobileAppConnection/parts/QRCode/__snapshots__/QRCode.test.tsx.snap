// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCode it renders properly 1`] = `
<div>
  <div
    class="root root_bordered"
  >
    <svg
      height="256"
      viewBox="0 0 21 21"
      width="256"
    >
      <path
        d="M0,0 h21v21H0z"
        fill="#FFFFFF"
        shape-rendering="crispEdges"
      />
      <path
        d="M0 0h7v1H0zM8 0h1v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM9 1h1v1H9zM11 1h2v1H11zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM9 2h3v1H9zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM10 3h1v1H10zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h3v1H8zM12 4h1v1H12zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM8 5h3v1H8zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM1 8h7v1H1zM11 8h2v1H11zM15 8h2v1H15zM20,8 h1v1H20zM0 9h2v1H0zM3 9h1v1H3zM8 9h1v1H8zM12 9h7v1H12zM20,9 h1v1H20zM3 10h1v1H3zM5 10h2v1H5zM9 10h1v1H9zM11 10h1v1H11zM13 10h2v1H13zM17 10h3v1H17zM0 11h1v1H0zM4 11h2v1H4zM7 11h1v1H7zM12 11h1v1H12zM16 11h3v1H16zM1 12h2v1H1zM4 12h1v1H4zM6 12h4v1H6zM11 12h1v1H11zM16 12h1v1H16zM20,12 h1v1H20zM8 13h3v1H8zM12 13h6v1H12zM20,13 h1v1H20zM0 14h7v1H0zM8 14h1v1H8zM12 14h1v1H12zM14 14h1v1H14zM18 14h2v1H18zM0 15h1v1H0zM6 15h1v1H6zM8 15h2v1H8zM13 15h2v1H13zM16 15h3v1H16zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM8 16h3v1H8zM16 16h1v1H16zM20,16 h1v1H20zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM8 17h2v1H8zM12 17h1v1H12zM15 17h3v1H15zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h2v1H8zM11 18h1v1H11zM13 18h1v1H13zM18 18h1v1H18zM0 19h1v1H0zM6 19h1v1H6zM8 19h1v1H8zM16 19h3v1H16zM0 20h7v1H0zM11 20h1v1H11zM13 20h1v1H13zM16 20h1v1H16zM19 20h1v1H19z"
        fill="#000000"
        shape-rendering="crispEdges"
      />
    </svg>
  </div>
</div>
`;
